### PR TITLE
Clear reloadable flag for workspace modules that happen to be dependencies of non-reloadable dependencies

### LINKED ISSUE #24895: Classloader Issues on MultiModule Project Tests
### Describe the bug

A Multimodule project in test mode gets classcast exceptions injecting beans. This started in 2.7.0 +

Works fine in 2.6.*

The specific case is there the multi module project has a library which is used in 2 modules - one of which is in the project itself and another is just included in the quarkus package.

### Expected behavior

Tests should pass

### Actual behavior

Tests fails due to class cast exceptions

### How to Reproduce?

git clone https://github.com/bcluap/quarkus-examples.git
cd quarkus-examples/multi-module/mm-2
mvn clean install
cd ..
mvn clean verify

If you change quarkus version in the mm-2 pom and main pom to 2.6.3.Final then the projects tests work fine.

If you add mm-2 as a module in MultiModule then the tests pass but that should not be necessary.

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by aloubyansky**
Fix #24895

This change makes sure that local project dependencies are not flagged as reloadable if they happen to be dependencies (direct or transitive) of non-reloadable dependencies.

### CODE REVIEW

**Comment by aloubyansky** [comment]
@gsmet @geoand I know it's hard to follow these changes, I'd like to get this in ASAP though to get this tested more before the next release.

**Review by geoand — APPROVED** [review]
Let's get it in!
